dm/syncer/validate_worker_test.go

```diff
@@ -0,0 +1,380 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncer
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb-tools/pkg/filter"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/parser/mysql"
+	"github.com/pingcap/tidb/parser/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pingcap/tiflow/dm/dm/pb"
+	"github.com/pingcap/tiflow/dm/pkg/conn"
+)
+
+// split into 3 cases, since it may be unstable when put together.
+func TestValidatorWorkerRunInsertUpdate(t *testing.T) {
+	tbl1 := filter.Table{Schema: "test", Name: "tbl1"}
+	tbl2 := filter.Table{Schema: "test", Name: "tbl2"}
+	tbl3 := filter.Table{Schema: "test", Name: "tbl3"}
+	tableInfo1 := genValidateTableInfo(t, tbl1.Schema, tbl1.Name,
+		"create table tbl1(a int primary key, b varchar(100))")
+	tableInfo2 := genValidateTableInfo(t, tbl2.Schema, tbl2.Name,
+		"create table tbl2(a varchar(100) primary key, b varchar(100))")
+	tableInfo3 := genValidateTableInfo(t, tbl3.Schema, tbl3.Name,
+		"create table tbl3(a varchar(100) primary key, b varchar(100))")
+
+	cfg := genSubtaskConfig(t)
+	_, mock, err := conn.InitMockDBFull()
+	mock.MatchExpectationsInOrder(false)
```

Comment:
why need add this?

Reply:
default, sqlmock match sqls in the order we write in code. but in validation it's iterating over a map, the order of sql may different, it's ok in `debug mode`, but unstable in `run mode`